src/main/java/software/amazon/cloudformation/Wrapper.java

```diff
@@ -0,0 +1,567 @@
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.HandlerRequest;
+import software.amazon.cloudformation.proxy.LoggerProxy;
+import software.amazon.cloudformation.proxy.MetricsPublisherProxy;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.WaitStrategy;
+import software.amazon.cloudformation.resource.ResourceTypeSchema;
+import software.amazon.cloudformation.resource.SchemaValidator;
+import software.amazon.cloudformation.resource.Serializer;
+import software.amazon.cloudformation.resource.Validator;
+import software.amazon.cloudformation.resource.exceptions.ValidationException;
+
+public abstract class Wrapper<ResourceT, CallbackT> {
+
+    public static final SdkHttpClient HTTP_CLIENT = ApacheHttpClient.builder().build();
+
+    private static final List<Action> MUTATING_ACTIONS = Arrays.asList(Action.CREATE, Action.DELETE, Action.UPDATE);
+
+    protected final Serializer serializer;
+    protected LoggerProxy loggerProxy;
+    protected MetricsPublisherProxy metricsPublisherProxy;
+
+    protected LoggerProxy platformLoggerProxy;
+    protected LogPublisher platformLogPublisher;
+
+    // provider... prefix indicates credential provided by resource owner
+    protected final CredentialsProvider providerCredentialsProvider;
+
+    protected final CloudWatchProvider providerCloudWatchProvider;
+    protected final CloudWatchLogsProvider cloudWatchLogsProvider;
+    protected final SchemaValidator validator;
+    protected final TypeReference<HandlerRequest<ResourceT, CallbackT>> typeReference;
+
+    protected MetricsPublisher providerMetricsPublisher;
+
+    protected CloudWatchLogHelper cloudWatchLogHelper;
+    protected CloudWatchLogPublisher providerEventsLogger;
+
+    protected Wrapper() {
+        this.providerCredentialsProvider = new SessionCredentialsProvider();
+        this.providerCloudWatchProvider = new CloudWatchProvider(this.providerCredentialsProvider, HTTP_CLIENT);
+        this.cloudWatchLogsProvider = new CloudWatchLogsProvider(this.providerCredentialsProvider, HTTP_CLIENT);
+        this.serializer = new Serializer();
+        this.validator = new Validator();
+        this.typeReference = getTypeReference();
+        this.platformLoggerProxy = new LoggerProxy();
+    }
+
+    /*
+     * This .ctor provided for testing
+     */
+    public Wrapper(final CredentialsProvider providerCredentialsProvider,
+                   final LogPublisher platformEventsLogger,
+                   final CloudWatchLogPublisher providerEventsLogger,
+                   final MetricsPublisher providerMetricsPublisher,
+                   final SchemaValidator validator,
+                   final Serializer serializer,
+                   final SdkHttpClient httpClient) {
```

Comment:
Why we need to add this `SdkHttpClient` as one of Wrapper constructor?   We are using a shared the HTTP client for all client to share connection pool etc.  It should be just to use a default static HTTP client.

Reply:
I did not touch this code, I just copied it from the existing lambdaWrapper. From what I could tell, this was done only so that we could mock the httpClient during testing easily